helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -376,62 +614,158 @@ public Object removeRoutingTableChangeListener(
    * @return
    */
   public Collection<LiveInstance> getLiveInstances() {
-    return _routingTableRef.get().getLiveInstances();
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getLiveInstances();
+  }
+
+  public Collection<LiveInstance> getLiveInstances(PropertyType propertyType) {
```

Comment:
This, I don't understand. Why live instance would be different with different state?

Reply:
Well, I understand they are the same. However, we need a key for accessing routingTableRefMap. Should I randomly pick one of the keys? It doesn't hurt having this method. I removed one of the methods. Please let me know if that answers your question.